lib/matplotlib/axes/_axes.py

```
@@ -4260,24 +4257,22 @@ def invalid_shape_exception(csize, nsize):
                             "with a single row if you really want to specify "
                             "the same RGB or RGBA value for all points.")
                     valid_shape = False
-                    c_array = None
-        if c_array is None:
+        if not c_is_mapped:
             try:  # Is 'c' acceptable as PathCollection facecolors?
                 colors = mcolors.to_rgba_array(c)
             except ValueError:
                 if not valid_shape:
-                    raise invalid_shape_exception(csize, xsize)
+                    raise invalid_shape_exception(c.size, xsize)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)